.github/workflows/publish-static.yml

```diff
@@ -0,0 +1,37 @@
+name: Publish static files to GitHub Pages
+
+# Workflow runs on pushes to the main branch. It publishes statics files to
+# GitHub Pages. The specs themselves (WebGPU, WGSL, the explainer) are handled
+# separately in a different workflow.
+# The "workflow_dispatch" hook allows admins to trigger the workflow manually
+# from GitHub's UI.
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Copy static files to the "out" folder
+        run: |
+          export PATH="$(python3 -m site --user-base)/bin:${PATH}"
+          ./compile.sh static
```

Comment:
It looks like the deployment depends on the spec HTML files, but this call does not build them. ?

Reply:
What do you mean by "depends on"? The `publish-static.yml` workflow is not supposed to depend on the spec HTML files.

The publication mechanism is split into two workflows:
- `publish-static.yml`: Publishes all files to GitHub Pages, except HTML files. `compile.sh` is used to copy all files except HTML files to the `out` folder.
- `publish-specs.yml`: Builds and publishes HTML files to GitHub Pages, and to /TR, using the w3c/spec-prod action. `compile.sh` is not used at all for that.

The `static` argument I introduced to the `compile.sh` script makes it bypass the generation of the HTML files. I did not modify the default behavior of `compile.sh` when called without arguments, both because the preview workflow depends on it, and because I thought that spec editors may depend on it to build the HTML files locally.

I could perhaps try to combine both jobs into the same workflow. I kept them separated to be able to run the `publish-specs.yml` workflow only on updates that modify the source of the HTML documents.

Reply:
Ok, I understand now.
I misunderstood the meaning of the CLEAN-EXCLUDE option and it set me off in the wrong direction.

Thanks for the explanation!

Reply:
I don't think I understand. We used to have the `publish` job that builds HTML and uploads them to pages. Now it's replaced by `publish-static`, which no longer builds HTML. What am I missing?

Reply:
> I don't think I understand. We used to have the `publish` job that builds HTML and uploads them to pages. Now it's replaced by `publish-static`, which no longer builds HTML. What am I missing?

You used to have **one** `publish` job that built HTML files and uploaded both static files and HTML files to GitHub Pages. That PR would replace this job with **two** jobs:
- one that publishes all static files, excluding HTML files, to GitHub Pages: `publish-static`
- one that publishes HTML files to GitHub Pages **and** to /TR: `publish-specs`

The second job is based on the w3c/spec-prod action. That action handles building the spec with Bikeshed, deploying the HTML page to GitHub Pages and publishing the HTML page to /TR. Problem is that this action does not handle static file at all (it might in the future, see https://github.com/w3c/spec-prod/issues/39). This is why I need to keep a second job that deploys static files to GitHub Pages for now.

I'm viewing that deploy workflow from a `w3c/spec-prod` + static files perspective. That is, my starting point here was the `w3c/spec-prod` action and I was not necessarily trying to preserve the current workflow.

I suspect that you're looking at this PR expecting to see an update that would keep the existing workflow and complete it with deployment to /TR. If that seems needed, that is possible: we could keep the existing `publish` job intact and add a "deploy to /TR" job that uses the `w3c/spec-prod` action but would no longer publish to GitHub Pages and would only build the HTML pages to publish to /TR.

Reasons why I favored an approach based on `w3c/spec-prod`:
- I hope that this will be become the default action to build and deploy all specs in the future, and that the action will also handle static files. In other words, I'm hoping that we can eventually drop the `publish-static` job altogether.
- It seems good to use only one way to build HTML files. The `compile.sh` script and the `w3c/spec-prod` action currently install Bikeshed the same way, but it seems possible that the two scripts no longer use exactly the same version of Bikeshed or no longer call Bikeshed with the exact same parameters in the future, which could perhaps create hard to spot problems. In other words, boilerplate excepted, I'd like to make sure that the ED and the /TR versions are always exactly the same.

Reply:
If webgpu.idl were the only static file, I'd say we could just drop it and simplify a lot. But right now we export the samples as well, so I think this is the right way to do it. Thank you for figuring this out, it must have been a challenge (had to find that CLEAN-EXCLUDE option!)

Reply:
Regarding `webgpu.idl`, I do not know what use cases the file is meant to cover. If the goal is to have an always-up-to-date IDL extract, it should indeed be kept around. If the goal is rather to have an always-valid IDL extract, then as long as we don't have a "validate IDL" workflow at hand, relying on the `@webref/idl` package (see https://github.com/gpuweb/cts/issues/332#issuecomment-844893741 for details) might be better and the `webgpu.idl` file could perhaps be dropped.

Experience with integrating IDL extracts in other tools (Web Platform Tests, but also [types-web](https://github.com/saschanaz/types-web#types-web) or [WebIDLPedia](https://dontcallmedom.github.io/webidlpedia/)) suggests that it is more important to have an extract that comes with guarantees than an extract that you may not even be able to parse.

Reply:
We put it there because it is occasionally convenient to be able to get previous versions of the idl without rebuilding, but it's not necessary. We _really_ included it just because we were generating it already and it was trivial to include along with the build.

The IDL should always be at least parseable, as Bikeshed does parse it.